hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/FileSinkBuilder.java

```diff
@@ -96,6 +99,8 @@
      * Enables rolling by file size. If the size after writing a batch of items
      * exceeds the limit, a new file will be started. From this follows that
      * the file will typically be larger than the given maximum.
+     *
+     * @since 4.0
      */
     public FileSinkBuilder<T> rollByFileSize(@Nullable Long maxFileSize) {
```

Comment:
wouldn't it be better to take primitive `long`? what good does it to do pass null? You could use Long.MAX_VALUE if you don't want rolling by size.

Reply:
You should then provide a constant for the user, the user has to know it or has to read the javadoc. 

Also we add a suffix to the file name when we use rolling, Long.MAX_VALUE says "roll when you reach this impossible to reach size", and it's not the same as "don't use rolline".

I'm not strictly against, but seems clearer to me in current shape.

Reply:
I just find it a bit strange to pass `null`, since you're not required to call the method?